queue_services/entity-filer/src/entity_filer/filing_processors/filing_components/alternate_name.py

```diff
@@ -0,0 +1,190 @@
+            end_date=None,
+            identifier=legal_entity.identifier,
+            name=to_legal_name,
+            name_type=AlternateName.NameType.OPERATING,
+            start_date=change_filing.get("startDate"),
+            registration_date=change_filing_rec.effective_date,
+        )
+        legal_entity.alternate_names.append(new_alternate_name)
+
+        filing_meta = {
+            **filing_meta,
+            "fromLegalName": alternate_name.name,
+            "toLegalName": to_legal_name,
+        }
+
+    # Update Nature of LegalEntity
+    if (
+        naics := change_filing.get(f"{filint_type}", {})
+        .get("business", {})
+        .get("naics")
+    ) and (naics_code := naics.get("naicsCode")):
+        if legal_entity.naics_code != naics_code:
+            filing_meta = {
+                **filing_meta,
+                **{
+                    "fromNaicsCode": legal_entity.naics_code,
+                    "toNaicsCode": naics_code,
+                    "naicsDescription": naics.get("naicsDescription"),
+                },
+            }
+            legal_entity_info.update_naics_info(legal_entity, naics)
+
+
+def update_proprietor_change(
+        filint_type: str,
+        change_filing_rec: Filing,
+        change_filing: Dict,
+        filing_meta: Dict,
+):
+    name_request = dpath.util.get(change_filing, f"/{filint_type}/nameRequest", default=None)
+    identifier = dpath.util.get(change_filing_rec.filing_json, "filing/business/identifier")
+    if name_request and (to_legal_name := name_request.get("legalName")):
+        alternate_name = AlternateName.find_by_identifier(identifier)
+        parties_dict = dpath.util.get(change_filing, f"/{filint_type}/parties")
+
+        # Find the Proprietor
+        proprietor = None
+        for party in parties_dict:
+            for role in party.get("roles"):
+                if role.get("roleType") == "Proprietor":
+                    proprietor_dict = party
+                    break
+            if proprietor_dict:
+                break
+
+        if not proprietor_dict:
+            raise DefaultException(
+                f"No Proprietor in the SP {filint_type} for filing:{change_filing_rec.id}"
+            )
+
+        proprietor, delivery_address, mailing_address = get_or_create_party(
+            proprietor_dict, change_filing_rec
+        )
+        if not proprietor:
+            raise DefaultException(
+                f"No Proprietor in the SP {filint_type} for filing:{change_filing_rec.id}"
+            )
+        
+        if start := change_filing.get("filing", {}).get(f"{filint_type}", {}).get("startDate"):
+            start_date = LegislationDatetime.as_utc_timezone_from_legislation_date_str(
+                start
+            )
+        elif change_filing.effective_date:
+            start_date = change_filing.effective_date.isoformat()
+        else:
+            start_date = LegislationDatetime.now()
```

Comment:
Did you copy this from registration? This is not correct. In change of registration start date cannot be updated, so keep the existing value (`start_date=alternate_name.start_date,`) and in conversion it is optional to update start date. Update this accordingly

Reply:
updated